### PR TITLE
Added `output_processor` parameter to `BashProcessor`

### PR DESCRIPTION
**Pull request by dabla**
Added result_processor parameter to BashProcessor which allows you to specify a lambda function that processes the output of the bash script before it is pushed as an XCom. This feature is particularly useful for manipulating the script’s result directly within the BashOperator, without the need for additional operators or tasks.

If for example you have following bash script which uses the linux [jq ](https://jqlang.github.io/jq/) command to filter the required elements from a JSON-file, it is handy to have the XCom directly loaded as a JSON object in python instead of a string:

```
filter_today_changes = BashOperator(
        task_id="filter_today_changes",
        bash_command="""
            jq -c '.[] | select(.lastModified > "{{ data_interval_start | ts_zulu }}" or .created > "{{ data_interval_start | ts_zulu }}")' \\
           example.json
        """,
        output_processor=lambda output: json.loads(output),
)
```

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/operators/bash.py [hunk 1]
```diff
@@ -63,6 +63,8 @@ class BashOperator(BaseOperator):
         If None (default), the command is run in a temporary directory.
         To use current DAG folder as the working directory,
         you might set template ``{{ dag_run.dag.folder }}``.
+    :param result_processor: Function to further process the result of the bash script
+        (default is lambda: result: result).
```
Comment (shahar1) [inline airflow/operators/bash.py]:
```suggestion
        (default is lambda result: result).
```

File: airflow/operators/bash.py [hunk 2]
```diff
@@ -63,6 +63,8 @@ class BashOperator(BaseOperator):
         If None (default), the command is run in a temporary directory.
         To use current DAG folder as the working directory,
         you might set template ``{{ dag_run.dag.folder }}``.
+    :param result_processor: Function to further process the result of the bash script
```
Comment (shahar1) [inline airflow/operators/bash.py]: I'd rather call it `output_processor` to be consistent with terminology (i.e., existing `output_encoding`)

File: docs/apache-airflow/howto/operator/bash.rst [hunk 3]
```diff
@@ -184,6 +184,48 @@ exit code if you pass ``skip_on_exit_code``).
             :end-before: [END howto_operator_bash_skip]
 
 
+Result processor
+----------------
+
+The result_processor parameter allows you to specify a lambda function that processes the output of the bash script
```
Comment (shahar1) [inline docs/apache-airflow/howto/operator/bash.rst]:
```suggestion
The ``result_processor`` parameter allows you to specify a lambda function that processes the output of the bash script
```

File: airflow/operators/bash.py [hunk 4]
```diff
@@ -243,7 +250,7 @@ def execute(self, context: Context):
                 f"Bash command failed. The command returned a non-zero exit code {result.exit_code}."
             )
 
-        return result.output
+        return self.result_processor(result.output)
```
Comment (shahar1) [inline airflow/operators/bash.py]: Could you please add a unit test?

**Review by shahar1 — COMMENTED** [review]
Seems like a useful feature, got some few comments :)

File: airflow/operators/bash.py (hunk 4 above)
Comment (shahar1) [reply airflow/operators/bash.py]: Still relevant :)

File: airflow/operators/bash.py (hunk 2 above)
Comment (shahar1) [reply airflow/operators/bash.py]: I'm a bit confused - do you agree with the renaming? (it is not implemented in practice)

Comment (dabla) [reply airflow/operators/bash.py]: I thought I've pushed my changes but didn't, now it should be ok

Comment (dabla) [reply airflow/operators/bash.py]: I completely agree on the renaming ;)

File: airflow/operators/bash.py [hunk 5]
```diff
@@ -130,6 +132,9 @@ class BashOperator(BaseOperator):
             env={"message": '{{ dag_run.conf["message"] if dag_run else "" }}'},
         )
 
+    .. versionadded:: 2.10.0
+       The `result_processor` parameter.
```
Comment (shahar1) [inline airflow/operators/bash.py]:
```suggestion
       The `output_processor` parameter.
```

File: docs/apache-airflow/howto/operator/bash.rst [hunk 6]
```diff
@@ -184,6 +184,48 @@ exit code if you pass ``skip_on_exit_code``).
             :end-before: [END howto_operator_bash_skip]
 
 
+Result processor
```
Comment (shahar1) [inline docs/apache-airflow/howto/operator/bash.rst]: Don't forget updating the docs :)
(see also `result_processor` references below)

**Review by shahar1 — COMMENTED** [review]
Almost there

**Review by potiuk — APPROVED** [review]

File: docs/apache-airflow/howto/operator/bash.rst (hunk 6 above)
Comment (potiuk) [reply docs/apache-airflow/howto/operator/bash.rst]: Yeah the example in the doc should be corrected :)

Comment (dabla) [reply docs/apache-airflow/howto/operator/bash.rst]: Just did it

**Pull request merged**